bsp4j/src/main/java/ch/epfl/scala/bsp4j/BuildClient.java

```diff
@@ -0,0 +1,35 @@
+package ch.epfl.scala.bsp4j;
+
+import org.eclipse.lsp4j.jsonrpc.services.JsonNotification;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface BuildClient {
+
+    @JsonNotification("build/showMessage")
+    void showMessage(ShowMessageParams params);
+
+    @JsonNotification("build/logMessage")
+    void logMessage(LogMessageParams params);
+
+    @JsonNotification("build/publishDiagnostics")
+    void publishDiagnostics(PublishDiagnosticsParams params);
+
+    @JsonNotification("buildTarget/didChange")
+    void didChangeBuildTarget(DidChangeBuildTarget params);
+
+    @JsonRequest("build/registerFileWatcher")
+    CompletableFuture<RegisterFileWatcherResult> registerFileWatcher(RegisterFileWatcherParams params);
+
+    @JsonRequest("build/cancelFileWatcher")
+    CompletableFuture<CancelFileWatcherResult> cancelFileWatcher(CancelFileWatcherParams params);
+
+    @JsonNotification("buildTarget/compileReport")
+    void compileReport(CompileReport params);
```

Comment:
lsp4j uses a feature called "segments" so that all `buildTarget/*` methods are defined in shared trait like `BuildTargetService`

```scala
trait BuildServer {
  def getWorkspaceService: WorkspaceService
  def getBuildService: BuildService
  def getBuildTargetService: BuildTargetService
}
trait WorkspaceService {
  def buildTarget()
}
trait BuildService {
  def registerFileWatcher()
}
trait BuildTargetService {
  def compileReport(..)
}
```

Although this pleases my OCD by being consistent with the protocol I found it awkward to use in practice compared to a flat list of methods.